movement/filesystem.c

```diff
@@ -166,6 +166,28 @@ bool filesystem_read_file(char *filename, char *buf, int32_t length) {
     return false;
 }
 
+bool filesystem_read_line(char *filename, char *buf, int32_t offset, int32_t length) {
+    memset(buf, 0, length);
+    int32_t file_size = filesystem_get_file_size(filename);
+    if (file_size > 0) {
+        int err = lfs_file_open(&lfs, &file, filename, LFS_O_RDONLY);
+        if (err < 0) return false;
+        err = lfs_file_seek(&lfs, &file, offset, LFS_SEEK_SET);
+        if (err < 0) return false;
+        err = lfs_file_read(&lfs, &file, buf, min(length, file_size - offset));
+        if (err < 0) return false;
+        for(int i = 0; i < length; i++) {
+            if (buf[i] == '\n') {
+                buf[i] = 0;
+                break;
+            }
+        }
```

Comment:
I feel like there's a subtle bug here where you don't get null terminated string if you read length and don't hit a new line or if your file is less than length and it doesn't end in a newline

Reply:
Good call; it works as long as you have a buffer one byte longer than the maximum length you're requesting, but it's probably safer to just max out at `length - 1`.